### PR TITLE
Registration, email automation, history, & bulk edit

### PR DESCRIPTION
**Pull request by JoelForamitti**
- Adds a membership registration form
- Adds an email automation function
- Adds history viewsets with a revert function
- Adds bulk edit function

### CODE REVIEW

File: collectivo/profiles/serializers.py [hunk 1]
```diff
@@ -2,25 +2,76 @@
...
+            "label": "Address",
+            "fields": [
+                "address_street",
+                "address_number",
+                "address_stair",
+                "address_door",
+                "address_postcode",
+                "address_city",
+                "address_country",
+                "phone",
```
Comment (nanu-c) [inline collectivo/profiles/serializers.py:70]: Please divide in 3 sections:
                "address_street",
                "address_number",
                "address_stair",
                "address_door",
and 
                "address_postcode",
                "address_city",
                "address_country",
and 
"phone"

**Review by nanu-c — CHANGES_REQUESTED** [review]

File: collectivo/profiles/serializers.py (hunk 1 above)
Comment (JoelForamitti) [reply collectivo/profiles/serializers.py]: done

File: collectivo/memberships/tests.py [hunk 2]
```diff
@@ -86,21 +70,24 @@ def test_automatic_emails(self, chain):
...
-        self.membership.save()
+        # self.membership.date_ended = "2020-01-01"
+        # self.membership.save()
 
-        run_mocked_celery_chain(chain)
-        self.assertEqual(len(mail.outbox), 3)
-        self.assertEqual(mail.outbox[2].subject, "Test Subject Ended")
+        # run_mocked_celery_chain(chain)
+        # self.assertEqual(len(mail.outbox), 3)
+        # self.assertEqual(mail.outbox[2].subject, "Test Subject Ended")
```
[the remark this answers is not in the document]
Comment (JoelForamitti) [reply collectivo/memberships/tests.py:91]: done

File: collectivo/core/models.py [hunk 3]
```diff
@@ -97,11 +99,28 @@ class PermissionGroup(models.Model):
...
 
     class Meta:
         """Model settings."""
 
         unique_together = ("name", "extension")
+
+    def __str__(self):
+        """Return the string representation."""
+        if self.extension:
+            return f"{self.extension} {self.name}"
```
[the remark this answers is not in the document]
Comment (JoelForamitti) [reply collectivo/core/models.py:125]: done

**Review by nanu-c — APPROVED** [review]

**Pull request merged**